idea/src/org/jetbrains/kotlin/idea/inspections/KotlinRedundantOverrideInspection.kt

```diff
@@ -98,4 +105,21 @@ class KotlinRedundantOverrideInspection : AbstractKotlinInspection(), CleanupLoc
     companion object {
         private val MODIFIER_EXCLUDE_OVERRIDE = KtTokens.MODIFIER_KEYWORDS_ARRAY.asList() - KtTokens.OVERRIDE_KEYWORD
     }
-}
\ No newline at end of file
+}
+
+private fun KtNamedFunction.isDefinedInDelegatedSuperType(): Boolean {
+    val context = analyze()
+    return containingClassOrObject?.superTypeListEntries?.any { entry ->
+        val superType = (entry as? KtDelegatedSuperTypeEntry)?.typeReference?.let { context[BindingContext.TYPE, it] }
+        val superTypeDescriptor = superType?.constructor?.declarationDescriptor as? ClassDescriptor
+        superTypeDescriptor?.let { isDefinedIn(listOf(it)) } ?: false
+    } ?: false
+}
+
+private fun KtNamedFunction.isDefinedIn(classDescriptors: List<ClassDescriptor>): Boolean {
+    val functionName = name ?: return false
+    return classDescriptors.any {
+        it.findDeclaredFunction(functionName, false) { it.valueParameters == valueParameters } != null
+        || isDefinedIn(it.getSuperInterfaces())
```

Comment:
Here it's also not clear why super interfaces only and not the super class

Reply:
I think that checking super classes is unnecessary. A class delegation can only use interface and an interface can not inherit from a class.